src/shared/parameter_v2/scripts/config_yaml_loader.py

```diff
@@ -0,0 +1,288 @@
+import os
+import sys
+
+import networkx
+import yaml
+from colorama import Fore, init
+import jsonschema
+
+from dynamic_parameter_schema import (
+    INCLUDE_DEF_SCHEMA,
+    PARAM_DEF_SCHEMA,
+    SUPPORTED_TYPES,
+)
+
+#######################################################################
+#                         Config Yaml Loader                          #
+#######################################################################
+
+
+class ConfigYamlLoader(object):
+    @staticmethod
+    def get_config_metadata(yaml_paths: list):
+        """Loads the yamls in the root config directory and verifies that the
+        requested configuration is valid. Then returns config_metadata dict
+        with a specified format capturing all the requested params and configs.
+
+        Example config directory:
+        .
+        └── config
+            ├── ai
+            │   ├── ai.yaml
+            │   ├── passing
+            │   │   └── passing.yaml
+            │   └── plays
+            │       ├── plays.yaml
+            │       ├── defense.yaml
+            │       └── offense.yaml
+            ├── sensor_fusion
+            │   └── sensor_fusion.yaml
+            └── tbots_config.yaml
+
+        A valid config directory maintains the following properties:
+
+        - All yamls are properly formed (i.e proper syntax, YAML Version 1.2)
+        - All parameter definitions abide by the dynamic parameter schema
+        - All include statements do NOT cause cycles in configs
+
+        :param yaml_paths: the path to all the config yamls
+        :type yaml_paths: list of str
+
+        :returns config_metadata: the config name followed by a list of
+            included configs and a list of defined parameters.
+        {
+            "ai.yaml": {
+                "includes": ["passing.yaml", "plays.yaml"],
+                "parameters" :
+                [{
+                    "int": {
+                        "name": "example_int", "value": 10, "min": 0, "max": 20
+                    },
+                    ...
+                }]
+            },
+
+            "plays.yaml": {
+                "includes": ["defense.yaml", "offense.yaml"],
+                ...
+            }
+            ...
+        }
+
+        :rtype: dict
+
+        """
+        # load yaml into config_metadata
+        config_metadata = ConfigYamlLoader.__load_yaml_into_dict(yaml_paths)
+
+        # validate schema
+        ConfigYamlLoader.__validate_config_metadata(config_metadata)
+
+        # detect cycles
+        ConfigYamlLoader.__detect_cycles_in_config_metadata(config_metadata)
+
+        # valid config ready to be generated
+        return config_metadata
+
+    @staticmethod
+    def __load_yaml_into_dict(yaml_paths: list):
```

Comment:
I think you can specify the type of a list's contents with `list(str)` or something similar?

Reply:
`list(str)` didn't seem to work

Do we have to do this?
~~~
from typing import List
YamlPathList = List[str]

def __load_yaml_into_dict(yaml_paths: YamlPathList):
~~~

Reply:
When you mean this, do you mean `YamlPathList` or `list`?

Reply:
Whoops, should've been more specific. I'm OK with the code snippet you posted above :stuck_out_tongue:

Reply:
Either is fine with me

Reply:
added `YamlPathList`